doc/api/packages.md

```diff
@@ -819,10 +747,213 @@ conditional exports for consumers could be to add an export, e.g.
 }
 ```
 
+## Node.js `package.json` field definitions
+
+This section describes the fields used by the Node.js runtime. Other tools (such
+as [npm](https://docs.npmjs.com/creating-a-package-json-file)) may use
+additional fields which are ignored by Node.js and not documented here.
+
+### `"name"`
+<!-- YAML
+added:
+  - v12.16.0
+  - v13.1.0
+changes:
+  - version:
+    - v12.16.0
+    - v13.6.0
+    pr-url: https://github.com/nodejs/node/pull/31002
+    description: Remove the `--experimental-resolve-self` option.
+-->
+
+```json
+{
+  "name": "package-name"
+}
+```
+
+* Type: {string}
+
+The `"name"` field defines your package’s name. Node.js doesn't apply any
+restriction on the name field, although the field is ignored if it is not a
+string or an empty string. Other tools (such as
+[npm](https://docs.npmjs.com/files/package.json#name)) may apply other
+restrictions on this field.
+
+The `"name"` field can be used in addition to the [`"exports"`][] field to
+[self-reference][] a package using its name.
+
+### `"type"`
+<!-- YAML
+added: v12.0.0
+changes:
+  - version:
+    - v12.17.0
+    - v13.2.0
+    pr-url: https://github.com/nodejs/node/pull/29866
+    description: Unflag `--experimental-modules`.
+-->
+
+* Type: {string}
+
+The `"type"` field defines the module format `.js` files should be treated
+within a particular `package.json` file’s package scope.
+
+Files ending with `.js` will be loaded as ES modules when the nearest parent
+`package.json` file contains a top-level field `"type"` with a value of
+`"module"`.
+
+The nearest parent `package.json` is defined as the first `package.json` found
+when searching in the current folder, that folder’s parent, and so on up
+until a node_modules folder or the volume root is reached.
+
+```json
+// package.json
+{
+  "type": "module"
+}
+```
+
+```console
+# In same folder as above package.json
+node my-app.js # Runs as ES module
+```
+
+If there is no package.json scope found, or if the nearest parent `package.json`
+lacks a `"type"` field, or contains `"type": "commonjs"`, or any other
+unsupported value, `.js` files are treated as CommonJS.
+
+```js
+// my-app.js, part of the same example as above
+import './startup.js'; // Loaded as ES module because of package.json
+```
+
+Regardless of the value of the `"type"` field, `.mjs` files are always treated
+as ES modules and `.cjs` files are always treated as CommonJS.
+
+### `"exports"`
```

Comment:
Is this whole section copied from `esm.md` but not deleted from the old file? I don't think we want two copies of it.

Reply:
We already moved all of esm over here. There is no exports section in esm anymore.

Reply:
Then why is there no large block of deleted lines in this PR where the previous `"exports"` section came from? Currently it looks like this PR adds the entire `"exports"` section.

Reply:
There is for "imports", but for "exports", this is replicating content from the upper guidance section. Again worth thinking about the reference v guide distinction. I still think reference here makes sense.

Reply:
There were no `"exports"` section in `esm.md`, the closest there was is the `Package entry point` section. This section is "new" content (or rather copied from #32970).